doc/source/introduction_intro.rst

```diff
@@ -14,49 +14,93 @@ to facilitate access to and exchange of genomics data across remote sites.
+
+The API consists of a series of :ref:`apidefinition`, or schemas, that define datasets, metadata, read group sets, reads, variants, etc. 
+
+The GA4GH web API schemas show developers how to connect data consumers ('clients') to data providers ('servers').
+They define how the data is organized, and thereby give information on what can be requested.
+The schemas are defined in Avro Interactive Data Language (extension .avdl), and 
+can be used to create code in any programming language.
 
 --------------------------
-The GA4GH web API
+API design patterns
 --------------------------
-This API was created to enable researchers to better access and exchange genomic data across remote sites. Instead of downloading complete BAM files or
-whole genome annotations, the API allows retrieval of information on, for instance, single genes or genomic regions.
+Web APIs use the protocols and conventions of the Web to connect consumers and providers of data, known as clients and servers.  This API uses these patterns:
 
-For a full list of the GA4GH API goals, see :ref:`apigoals`
+* HTTP protocol
+* ReST (Representational State Transfer) conventions
+* Paging
+* AVRO definition language
+* JSON representation
 
-The API consists of a series of :ref:`apidefinition`, or schemas, that define datasets, metadata, read group sets, reads, variants, etc. 
+For more details on web APIs, see `this wikipedia page <https://en.wikipedia.org/wiki/Web_API>`_  
 
-The schemas are written in Avro Interactive Data Language (extension .avdl). 
+HTTP protocol
+.............
+API clients use HTTP (`Hypertext Transfer Protocol <https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol>`_) for sending API queries and getting responses.
+HTTP is the same protocol that web browsers use for requesting pages from web servers.
 
-:ref:`avro` is a data serialization system, it defines how data gets transported across the internet.
-Here, data means both the request ('send me RNASeq for gene X in sample Y') and the response (the BAM file). Apache Avro only sends data in
-:ref:`json` or Binary Avro format, so requests and responses must be converted into one of those formats.
+In a typical conversation, a client sends a GET request to a server to ask for a specific resource, such as a set of reads.
+If the request contains appropriate identification and authorization information, the server responds with a copy of the requested information.
 
-For more details on AVRO, see :ref:`avro`
+Many GA4GH API servers will also support HTTPS, a variant of HTTP that uses encryption to protect messages from eavesdropping and modification while in transit.
 
-For more details on JSON, see :ref:`json`
+ReST (Representational State Transfer) conventions
+..................................................
+The GA4GH APIs follow `ReST conventions <https://en.wikipedia.org/wiki/Representational_state_transfer>`_ for web services on HTTP.  In particular:
 
```

Comment:
It's true that there's no spec and there are many arguments about what is and isn't REST.  But that doesn't strike me as a good argument for claiming that it is.  Anyways, here's some that come to mind:

* AFAIK, HTTP status codes are not well defined
* The use of POST everywhere frustrates the ability to use proxies for caching
* The API generally finds objects/data by searching as opposed to each object having a uniform locator
* Most REST API's that I've seen have a logical nesting of URI's for objects that are naturally part of an identifiable collection.  Our API smells nothing like that.
* HATEOAS - The documents the API produces should contain URIs that reference other objects in the system, not just ids that can be used in subsequent searches.  I think most people bend the rules on this one and are ok calling JSON "hypertext", but we don't use it that way.

Since it's mostly search-based I would propose that this API's current design isn't really amenable to REST, unlike a forum or something.